### ISSUE TITLE
vg and xg produce incompatible .xg formats despite using the same version of xg

### ISSUE THREAD

**Issue by ekg** [vgteam/vg#2442]
I was hoping to take an xg index made with vg and open it with the xg executable. However, I get a segfault.

```
vg augment -i call/HGSVC_chr22_17119590_17880307.vg call/HGSVC_chr22_17119590_17880307_alts.gam > HGSVC_alts.vg
vg index HGSVC_alts.vg -x HGSVC_alts.xg -L
xg -i HGSVC_alts.xg -G # segfaults
```

It seems that the wrapping container format in vg is causing this problem.

If we want to break apart vg, then we may need to consider dropping use of this secondary wrapper, or else we'll be pulling it in as a dependency wherever we read and write vg-related data formats in order to maintain compatibility with vg.

Another example is that we can no longer concatenate vg files using `cat`. The use of gzip by default also causes problems for performance, as things are basically limited by single-threaded compression and decompression. I think we should reconsider this.

**Comment by ekg** [comment]
My vote would be to stop using the container formatting. It's not clear what benefit it's providing, and it's breaking many flexible workflows.

**Comment by glennhickey** [comment]
I for one like the container format and would be in favour of keeping it, and making it easy to pull in as a dependency for any project that wants to write graphs.  I like it because it allows code like this to load a graph without caring about the underlying format
```
unique_ptr<HandleGraph> handle_graph = vg::io::VPKG::load_one<HandleGraph>(file_name);
```
In #2431, for example, I use this to let `vg pack` run on any kind of graph instead of just xg.  That's handy and is a cool way to leverage all the effort we've put into abstracting everything behind handle graphs. 

Also, I've never had problems with cat.  I don't think the toil-vg tests would pass if this broke.   The following code runs fine, for example: 
```
vg construct -v small/xy.vcf.gz -r small/xy.fa -R x > x.vg
vg construct -v small/xy.vcf.gz -r small/xy.fa -R y > y.vg
vg ids -j x.vg y.vg
vg stats -NE x.vg
vg stats -NE y.vg
cat x.vg y.vg | vg stats -NE -
```

**Comment by ekg** [comment]
I agree that this is a powerful system that can increase flexibility. My concern is that prevents the independence of any system that writes vg data formats from vg-specific libraries.

Now, everything that writes xg files will need vgpkg too or it will be unable to read output of the same format from vg. That's fine, but it adds complexity and dependencies to all these libraries.
